reference/test/stop/residual_norm_kernels.cpp

```diff
@@ -102,6 +103,67 @@ TYPED_TEST(ResidualNorm, CanCreateFactory)
     ASSERT_EQ(this->abs_factory_->get_executor(), this->exec_);
 }
 
+TYPED_TEST(ResidualNorm, CheckIfResZeroConverges)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using NormVector = typename TestFixture::NormVector;
+    using T = typename TestFixture::ValueType;
+    std::shared_ptr<gko::LinOp> mtx = gko::initialize<Mtx>({1.0}, this->exec_);
+    std::shared_ptr<gko::LinOp> rhs = gko::initialize<Mtx>({0.0}, this->exec_);
+    std::shared_ptr<gko::LinOp> x = gko::initialize<Mtx>({0.0}, this->exec_);
+    std::shared_ptr<gko::LinOp> res_norm =
+        gko::initialize<NormVector>({0.0}, this->exec_);
+
+    {
+        auto criterion = gko::stop::ResidualNorm<T>::build()
+                             .with_reduction_factor(r<T>::value)
+                             .on(this->exec_)
+                             ->generate(mtx, rhs, x.get(), nullptr);
+        constexpr gko::uint8 RelativeStoppingId{1};
+        bool one_changed{};
+        gko::array<gko::stopping_status> stop_status(this->exec_, 1);
+        stop_status.get_data()[0].reset();
```

Comment:
not important: since this is always the same except for the baseline, maybe there is some worth in making this a function? same for the checks

Reply:
done